lib/tasks/pkg.rake

```diff
@@ -23,6 +23,9 @@ namespace :pkg do
     ref = ENV['ref'] || 'HEAD'
     version = `git show #{ref}:VERSION`.chomp.chomp('-develop')
     raise "can't find VERSION from #{ref}" if version.length == 0
-    `git archive --prefix=foreman-#{version}/ #{ref} | bzip2 -9 > pkg/foreman-#{version}.tar.bz2`
+    `git archive --prefix=foreman-#{version}/ #{ref} > pkg/foreman-#{version}.tar`
+    `npm install`
+    `tar --transform 's,^,foreman-#{version}/,' -r -f pkg/foreman-#{version}.tar node_modules`
```

Comment:
On Wed, Jun 15, 2016 at 10:06 AM, Dominic Cleal <notifications@github.com>
wrote:

> In lib/tasks/pkg.rake
> <https://github.com/theforeman/foreman/pull/3433#discussion_r67109902>:
>
> > @@ -23,6 +23,9 @@ namespace :pkg do
> >      ref = ENV['ref'] || 'HEAD'
> >      version = `git show #{ref}:VERSION`.chomp.chomp('-develop')
> >      raise "can't find VERSION from #{ref}" if version.length == 0
> > -    `git archive --prefix=foreman-#{version}/ #{ref} | bzip2 -9 > pkg/foreman-#{version}.tar.bz2`
> > +    `git archive --prefix=foreman-#{version}/ #{ref} > pkg/foreman-#{version}.tar`
> > +    `npm install`
> > +    `tar --transform 's,^,foreman-#{version}/,' -r -f pkg/foreman-#{version}.tar node_modules`
>
> that implies that you sure run webpack compile prior to creating the tar.
>
> After, surely? Same as current asset precompilation.
>
yes, because otherwise node modules would not be pulled in, your options
are:
1. include node modules in the tar and compile the assets on build time.
2. compile the assets during tar (but don't include node_modules at all).
3. package all of the assets npm packages as build time dependency, and
compile the assets at build time.

is there a forth option?


How does including node_modules differ from how we pull in puppet modules
> to the installer?
>
> The installer tarballs should be fixed to not include modules, it
> shouldn't be held up as a good example.
>
> The Foreman tarball doesn't currently contain external gems, it shouldn't
> contain external npm packages either.
>
thats why I prefer option 2.

> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman/pull/3433/files/8f53a9c9d7358b0b91e136bc2010b1d69fd9fadd#r67109902>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABOx9V0bsp8E1UdZXr1qCb3s3Cxxj86ks5qL6RhgaJpZM4IJHWI>
> .
>